aws/resource_aws_transfer_user_test.go

```diff
@@ -0,0 +1,370 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/transfer"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSTransferUser_basic(t *testing.T) {
+	var conf transfer.DescribedUser
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:      func() { testAccPreCheck(t) },
+		IDRefreshName: "aws_transfer_user.foo",
+		Providers:     testAccProviders,
+		CheckDestroy:  testAccCheckAWSTransferUserDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSTransferUserConfig_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSTransferUserExists("aws_transfer_user.foo", &conf),
+					testAccMatchResourceAttrRegionalARN("aws_transfer_user.foo", "arn", "transfer", regexp.MustCompile(`user/.+`)),
+					resource.TestCheckResourceAttrPair(
+						"aws_transfer_user.foo", "server_id", "aws_transfer_server.foo", "id"),
+					resource.TestCheckResourceAttrPair(
+						"aws_transfer_user.foo", "role", "aws_iam_role.foo", "arn"),
+					resource.TestCheckResourceAttr(
+						"aws_transfer_user.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_transfer_user.foo", "tags.NAME", "tftestuser"),
+				),
+			},
+			{
+				ResourceName:      "aws_transfer_user.foo",
+				ImportState:       true,
+				ImportStateIdFunc: testAccAWSTransferUserImportStateIdFunc("aws_transfer_user.foo"),
+				// We do not have a way to align IDs since the Create function uses resource.PrefixedUniqueId()
```

Comment:
With the resource ID and `Read` function updates, the comments here and `ImportStateCheck` can be removed and it should instead use `ImportStateVerify: true`. 👍